Codecademy/discourse-codebytes-plugin
# improve codebyte tag replacement on save

## Overview
replaces regex with new (and heavily unit-tested) `findCodeByte` function. This also means I had to add jest to this repo, as qunit just doesn't work with importing single functions (lol), but this is probably a good thing.

There is currently one case that does not quite work correctly:
```
[codebyte]
[codebyte]
[/codebyte]
```
ie, an extra opening tag. This renders sanely but breaks "save to post", not sure how much we care though 🤷 

### PR Checklist
- [ ] Related to JIRA ticket: [REACH-1006](https://codecademy.atlassian.net/browse/REACH-1006)

### Testing Instructions
1. run discourse locally with this branch
2. make sure "save to post" functionality works with multiple codebytes
3. run `yarn test` and make sure the tests pass
